src/c++/perf_analyzer/genai-pa/genai_pa/parser.py

```diff
@@ -143,4 +209,26 @@ def parse_args(argv=None):
     add_dataset_args(dataset_group)
 
     args = parser.parse_args(argv)
+
+    # Concurrency and request rate are mutually exclusive
+    # TODO: Review if there is a cleaner way to do this with argparse
+    if args.concurrency is not None and args.request_rate is not None:
+        parser.error(
+            "Arguments --concurrency and --request_rate are mutually exclusive."
+        )
+
+    if args.concurrency is None and args.request_rate is None:
+        args.concurrency = 1
+        print(
+            "Neither --concurrency nor --request_rate provided. Setting concurrency to 1."
+        )
+
+    # Update GenAI-PA non-range attributes to range format for PA
+    for attr_key in ["concurrency", "request_rate"]:
+        attr_val = getattr(args, attr_key)
+        if attr_val is not None:
+            setattr(args, f"{attr_key}_range", f"{attr_val}:{attr_val}:{attr_val}")
+        delattr(args, attr_key)
+
+    args = argparse.Namespace(**{k: v for k, v in vars(args).items() if v is not None})
```

Comment:
Code comment? This line is pretty intense.

Reply:
Can you explain why they need to be pruned?

Reply:
Sure.
The way we are building the command for the subprocess call is to iterate over the available options and concatenating a string.
I am pruning the args that have None or False set because neither --abc=None or --xyz=false are compliant with PA.